server/taskflows/hpccloud/taskflow/nwchem.py

```diff
@@ -0,0 +1,290 @@
+from jsonpath_rw import parse
+from bson.objectid import ObjectId
+
+import cumulus.taskflow
+from cumulus.tasks.job import download_job_input_folders, submit_job
+from cumulus.tasks.job import monitor_job, monitor_jobs
+from cumulus.tasks.job import upload_job_output_to_folder
+from cumulus.tasks.job import terminate_job
+from cumulus.transport.files.download import download_path_from_cluster
+from girder.utility.model_importer import ModelImporter
+from girder.api.rest import getCurrentUser
+from girder.constants import AccessType
+from girder_client import GirderClient, HttpError
+
+from hpccloud.taskflow.utility import *
+
+class NwChemTaskFlow(cumulus.taskflow.TaskFlow):
+    """
+    {
+        "input": {
+            "folder": {
+                "id": <the id of the folder containing input files>
+            },
+            "geometry": {
+                "id": <the file id of the geometry file>
+            }
+        },
+        "output": {
+            "folder": {
+                "id": <id of folder to upload output into>
+            }
+        },
+        "cluster": {
+            "_id": <id of cluster to run on>
+        }
+    }
+    """
+    NWCHEM_IMAGE = {
+        'name': 'NWChem_ParaView-5.0.1',
+        'owner': '695977956746'
+    }
+
+    def start(self, *args, **kwargs):
+        user = getCurrentUser()
+        # Load the cluster
+        # TODO: should this be in a common class?
+        cluster_id = parse('cluster._id').find(kwargs)
+        if cluster_id:
+            cluster_id = cluster_id[0].value
+            model = ModelImporter.model('cluster', 'cumulus')
+            cluster = model.load(cluster_id, user=user, level=AccessType.ADMIN)
+            cluster = model.filter(cluster, user, passphrase=False)
+            kwargs['cluster'] = cluster
+
+        profile_id = parse('cluster.profileId').find(kwargs)
+        if profile_id:
+            profile_id = profile_id[0].value
+            model = ModelImporter.model('aws', 'cumulus')
+            profile = model.load(profile_id, user=user, level=AccessType.ADMIN)
+            kwargs['profile'] = profile
+
+        kwargs['image_spec'] = self.NWCHEM_IMAGE
+        kwargs['next'] = setup_input.s()
+
+        super(NwChemTaskFlow, self).start(
+            setup_cluster.s(
+                self, *args, **kwargs))
+
+    def terminate(self):
+        self.run_task(nwchem_terminate.s())
+
+    def delete(self):
+        for job in self.get('meta', {}).get('jobs', []):
+            job_id = job['_id']
+            client = _create_girder_client(
+            self.girder_api_url, self.girder_token)
+            client.delete('jobs/%s' % job_id)
+
+            try:
+                client.get('jobs/%s' % job_id)
+            except HttpError as e:
+                if e.status != 404:
+                    self.logger.error('Unable to delete job: %s' % job_id)
+
+
+# TODO: move to common class
+def _create_girder_client(girder_api_url, girder_token):
```

Comment:
We already have this function [here](https://github.com/Kitware/HPCCloud/blob/master/server/taskflows/hpccloud/taskflow/utility/__init__.py#L42-L46)